src/app/utils/helper.ts

```diff
@@ -0,0 +1,235 @@
+      postcss()
+        // @ts-ignore
+        .process(val, { parser: cssjs })
+        .then((code) => resolve(code.css))
+        .catch((err) => reject(err));
+    });
+  } catch (err) {
+    console.error("Couldn't convert JS to CSS.", err);
+    return '';
+  }
+};
+
+export const generateHtml = async (document: MockDocumentDataType) => {
+  const dataToConsider = document;
+  const documentName = dataToConsider.name;
+  const sectionData: any = dataToConsider.sections;
+
+  let headerGlobalCss: string = '';
+  let globalFontFamily: string = '';
+  let globalFontSize: string = '';
+
+  if (dataToConsider.styles.GLOBAL.backgroundColor) {
+    const backgroundColorJson = {
+      backgroundColor: dataToConsider.styles.GLOBAL.backgroundColor,
+    };
+
+    headerGlobalCss = await toCSS(JSON.stringify(backgroundColorJson));
+  }
+
+  if (dataToConsider.styles.GLOBAL.fontFamily) {
+    const fontFamilyJson = {
+      fontFamily: dataToConsider.styles.GLOBAL.fontFamily,
+    };
+
+    globalFontFamily = await toCSS(JSON.stringify(fontFamilyJson));
+  }
+
+  if (dataToConsider.styles.GLOBAL.fontSize) {
+    const fontSizeJson = {
+      fontSize: dataToConsider.styles.GLOBAL.fontSize,
+    };
+
+    globalFontSize = await toCSS(JSON.stringify(fontSizeJson));
+  }
+
+  let htmlData: string = '';
+  let headerHtml = '';
+  let firstColumnHtml: string[] = [];
+  let secondColumnHtml: string[] = [];
+  let imageUrl = '';
+
+  // iterate over the sections keys
+  const keys: string[] = Object.keys(sectionData);
+  keys.forEach((key: string) => {
+    const sectionDetails = sectionData[key];
+    const currSectioName = sectionDetails.sectionName;
+
+    const currSectionSchema = sectionMockJson.find(
+      (section) => section.name === currSectioName
+    );
+    if (!currSectionSchema) return;
+
+    // we also need to remove the <img tag from the titleComp as we'll be adding it via jspdf
+    currSectionSchema.titleComp = currSectionSchema.titleComp.replace(
+      /<img[^>]*>/g,
+      ''
+    );
+
+    // iterate over the keys to replace
+    allKeysToReplace.forEach((keyToReplace) => {
+      const keyToReplaceRegex = new RegExp('\\' + keyToReplace, 'g');
+
+      const keyToReplaceValue =
+        sectionDetails.body[0][keyToReplace.split('${this.')[1].split('}')[0]];
+
+      currSectionSchema.component = currSectionSchema.component.replace(
+        keyToReplaceRegex,
+        keyToReplaceValue
+      );
+
+      const keyToReplaceTitle =
+        sectionDetails.header[keyToReplace.split('${this.')[1].split('}')[0]];
+
+      if (keyToReplace === '${this.image}' && keyToReplaceTitle) {
+        imageUrl = keyToReplaceTitle;
+      }
+
+      currSectionSchema.titleComp = currSectionSchema.titleComp.replace(
+        keyToReplaceRegex,
+        keyToReplaceTitle
+      );
+    });
+
+    if (currSectionSchema.isStaticHeader) {
+      headerHtml += `
+        <div style="${headerGlobalCss}; padding: 15px 30px 15px 30px;">
```

Comment:
Can you move all of the hard coded `padding` values into re-usable strings (if they are duplicated)? Will make it easier to understand what the padding is being applied to (such as a section or header), and changing it in a single place is safer

Reply:
Fixed. Added padding constants at the top.